src/dscanner/analysis/run.d

```diff
@@ -256,6 +256,20 @@ bool analyze(string[] fileNames, const StaticAnalysisConfig config, string error
 		scope p = new Parser!ASTBase(ast_m, input, false);
 		p.nextToken();
 		ast_m.members = p.parseModule();
+		string moduleName;
+		if (p.md !is null)
+		{
+			import std.algorithm : map;
+			ASTBase.ModuleDeclaration md = *p.md;
+			
+			if (md.packages.length != 0)
+			{
+				moduleName = md.packages.map!(e => e.toString()).join(".").dup;
```

Comment:
We should be careful with all these duplications here. At some point they might kill the performance of our project. Now I don't know if this is needed or not, but in principle it shouldn't be needed. If it's an array that you need to pass, then you could just create a slice out of it by using the slice operator `[0 .. length]`.